File: VB/MsaglHelpers/RoutingHelper.vb
Imports DevExpress.Diagram.Core
Imports DevExpress.Diagram.Core.Routing
Imports Microsoft.Msagl.Core.Routing

Namespace MsaglHelpers
    Public NotInheritable Class RoutingHelper

        Private Sub New()
        End Sub

        Public Shared Function GetDiagramConnectorType(ByVal routingMode As EdgeRoutingMode) As ConnectorType
            If routingMode = EdgeRoutingMode.StraightLine Then
                Return ConnectorType.Straight
            End If
            If routingMode = EdgeRoutingMode.Spline OrElse routingMode = EdgeRoutingMode.SplineBundling OrElse routingMode = EdgeRoutingMode.SugiyamaSplines Then
                Return ConnectorType.Curved
            End If
            Return ConnectorType.RightAngle
        End Function
    End Class
End Namespace

File: CS/MsaglHelpers/RoutingHelper.cs
using DevExpress.Diagram.Core;
using DevExpress.Diagram.Core.Routing;
using Microsoft.Msagl.Core.Routing;

namespace MsaglHelpers {
    public static class RoutingHelper {
        public static ConnectorType GetDiagramConnectorType(EdgeRoutingMode routingMode) {
            if(routingMode == EdgeRoutingMode.StraightLine) {
                return ConnectorType.Straight;
            }
            if(routingMode == EdgeRoutingMode.Spline || routingMode == EdgeRoutingMode.SplineBundling || routingMode == EdgeRoutingMode.SugiyamaSplines) {
                return ConnectorType.Curved;
            }
            return ConnectorType.RightAngle;
        }
    }
}